### PR TITLE
feat(css-parser, formatter): custom properties and var function

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Implements #984 

- `var()` function; it's a different function that probably will get its own lint rules (like, for example, only one fallback property)
- `CSSCustomProperty`

## Test Plan

New tests for formatter and parser.

### CODE REVIEW

File: internal/formatter/test-fixtures/css/function/input.test.md [hunk 1]
```diff
@@ -38,6 +39,7 @@
 	padding: calc(calc(1px, 2px), 3px);
 	height: calc(1px, 2px, 3px);
 	background-color: rgb(0 128 255);
+	border: var(--foo, red, blue, calc(10px 1rem));
```
Comment (ematipico) [inline internal/formatter/test-fixtures/css/function/input.test.md:42]: @yeonjuan probably this is a bug, the `+` gets removed.

Comment (yeonjuan) [reply internal/formatter/test-fixtures/css/function/input.test.md:42]: @ematipico Yes, It seems a bug in parsing or formatting `CSSNumber`. I'll make a separate pr to handle it :)

File: internal/css-parser/index.ts [hunk 2]
```diff
@@ -965,7 +981,30 @@ export function parseFunction(parser: CSSParser): CSSFunction {
...
+					parser.unexpectedDiagnostic({
+						description: descriptions.CSS_PARSER.INVALID_CUSTOM_PROPERTY,
+					});
+				}
+				i++;
+				params.push(parsedValue);
+			} else {
+				params.push(parsedValue);
+			}
+		}
```
Comment (yeonjuan) [inline internal/css-parser/index.ts:996]:
```
if (parsedValue) {
	if (i === 0 && isVarFunction) {
    // ,,,
		i++;
	}
	params.push(parsedValue);
}
```
I think we can remove the `else` statement :)

**Review by yeonjuan — APPROVED** [review]
LGTM! I left a minor suggestion :)

**Pull request merged**